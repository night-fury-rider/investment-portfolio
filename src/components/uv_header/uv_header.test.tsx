import React from 'react';
import { shallow } from 'enzyme';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import UVHeader from './uv_header';


Enzyme.configure({ adapter: new Adapter() });

describe('UVHeader Component Suite', ()=> {

  const headerTitle = 'Test Title',
        centralTitle = 12345678,
        theme = 'success',
        alt='Yuvi',
        logoWidth = '100px',
        logoAlt = 'Yuvraj',
        logoFile = 'test.png',
        primaryWebsite = 'www.test.com',
        repositoryUrl = 'test20.com',
        repositoryHeight ='100px',
        repositoryLogo = 'test.png';

  test('has correct snapshot', ()=> {
    expect(<UVHeader title={headerTitle}
                     centralTitle={centralTitle}
                     theme={theme}
                     alt={alt}
                     primaryWebsite={primaryWebsite}
                     logoWidth={logoWidth}
                     logoAlt={logoAlt}
                     logoFile={logoFile}
                     repositoryUrl={repositoryUrl}
                     repositoryHeight={repositoryHeight}
                     repositoryLogo={repositoryLogo}/>).toMatchSnapshot();
  });

  test('renders with props provided', ()=> {
    let shallowComponent = shallow(<UVHeader title={headerTitle} />);
    expect(shallowComponent.text()).toContain(headerTitle);
    expect(shallowComponent.find('.uv-central-title').length).toEqual(0);
    expect(shallowComponent.find('.toolbar.' + theme).length).toEqual(0);
    expect(shallowComponent.find('.toolbar a.website-link').prop('href')).not.toBeDefined();

    expect(shallowComponent.find('.website-link img').prop('width')).not.toEqual(logoWidth);
    expect(shallowComponent.find('.website-link img').prop('alt')).not.toEqual(logoAlt);
    expect(shallowComponent.find('.website-link img').prop('src')).not.toEqual(logoFile);
    expect(shallowComponent.find('.repository-link img').children().length).toEqual(0);

    shallowComponent = shallow(<UVHeader title={headerTitle}
                                         centralTitle={centralTitle}
                                         theme={theme}
                                         alt={alt}
                                         primaryWebsite={primaryWebsite}
                                         logoWidth={logoWidth}
                                         logoAlt={logoAlt}
                                         logoFile={logoFile}
                                         repositoryUrl={repositoryUrl}
                                         repositoryHeight={repositoryHeight}
                                         repositoryLogo={repositoryLogo}/>);


    expect(shallowComponent.find('.uv-central-title').length).toEqual(1);

    expect(shallowComponent.find('.toolbar.' + theme).length).toEqual(1);
    expect(shallowComponent.find('.website-link img').prop('width')).toEqual(logoWidth);

    expect(shallowComponent.find('a.repository-link').prop('href')).toEqual(repositoryUrl);
    expect(shallowComponent.find('.repository-link img').prop('alt')).toEqual(alt);
    expect(shallowComponent.find('.repository-link img').prop('height')).toEqual(repositoryHeight);
    expect(shallowComponent.find('.repository-link img').prop('src')).toEqual(repositoryLogo);


    shallowComponent = shallow(<UVHeader title={headerTitle}
                                         centralTitle={'' + centralTitle}
                                         repositoryUrl={repositoryUrl}/>);

    expect(shallowComponent.find('.uv-central-title').text()).toEqual('' + centralTitle);
    expect(shallowComponent.find('.repository-link img').prop('height')).toEqual('40px');
    expect(shallowComponent.find('.repository-link img').prop('src')).toEqual('logo192.png');
  });

});
