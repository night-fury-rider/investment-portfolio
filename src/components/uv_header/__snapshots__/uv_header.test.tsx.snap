// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Suite has correct snapshot 1`] = `
<Memo(UVHeader)
  alt="Yuvi"
  centralTitle={12345678}
  logoAlt="Yuvraj"
  logoFile="test.png"
  logoWidth="100px"
  primaryWebsite="www.test.com"
  repositoryHeight="100px"
  repositoryLogo="test.png"
  repositoryUrl="test20.com"
  theme="success"
  title="Test Title"
/>
`;
